Neos.Utility.Lock/Classes/TYPO3/Flow/Utility/Lock/Factory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace TYPO3\Flow\Utility\Lock;
+
+/*
+ * This file is part of the TYPO3.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use TYPO3\Flow\Annotations as Flow;
+
+/**
+ * A general lock class.
+ *
+ * @api
+ */
+class Factory
+{
+    /**
+     * @param string $subject
+     * @param boolean $exclusiveLock TRUE to, acquire an exclusive (write) lock, FALSE for a shared (read) lock. An exclusive lock is the default.
+     * @return Lock
+     * @deprecated Use Factory::acquireCallback, deprecated in Flow 3.3
+     */
+    public static function acquire($subject, $exclusiveLock = true)
```

Comment:
Why do we have this new method that is deprecated right away? Can't we just offer acquireCallback as new method?